### PR TITLE
Survive `std::less<>` in TClassEdit.

### PR DESCRIPTION
**Pull request by Axel-Naumann**

### CODE REVIEW

File: core/foundation/src/TClassEdit.cxx [hunk 1]
```diff
@@ -771,7 +771,10 @@ static bool IsDefElement(const char *elementName, const char* defaultElementName
       size_t end = findNameEnd(c,pos);
 
       std::string keypart;
-      TClassEdit::GetNormalizedName(keypart,std::string_view(c.c_str()+pos,end-pos));
+      if (pos != end) {
+         // ROOT-11000: elementName = "std::less<>"
```
Comment (pcanal) [inline core/foundation/src/TClassEdit.cxx:775]: The comment is "misplaced".  Visually it looks like it explains the "next" line while it actually explains the previous line.
```suggestion
      if (pos != end) {  // ROOT-11000: elementName = "std::less<>"
```

File: core/foundation/src/TClassEdit.cxx [hunk 2]
```diff
@@ -832,6 +835,9 @@ void TClassEdit::GetNormalizedName(std::string &norm_name, std::string_view name
 {
    norm_name = std::string(name); // NOTE: Is that the shortest version?
 
+   if (name.empty())
+      return;
+
```
Comment (pcanal) [inline core/foundation/src/TClassEdit.cxx:840]:
```suggestion
   if (name.empty())
      return;

   norm_name = std::string(name); // NOTE: Is that the shortest version?

```

Comment (Axel-Naumann) [reply core/foundation/src/TClassEdit.cxx:839]: That would leave `norm_name` with whatever it had before:
```
std::string norm_name{"TObject"};
TClassEdit::NormalizeName(norm_name, "");
/// Now `norm_name` is "TObject"
```
Why would we want that?

Comment (pcanal) [reply core/foundation/src/TClassEdit.cxx:839]: right, I 'forgot' (did not see from the web interface) that norm_name was the output parameter.  Still, this alternative may (or may not) be 'faster' (depending how smart the optimizer is).
```suggestion
   if (name.empty()) {
      norm_name.clear();
      return;
   }
   
   norm_name = std::string(name); // NOTE: Is that the shortest version?
```

**Pull request merged**